### PR TITLE
Updating task Meow to support bot SSO

### PR DESCRIPTION
**Pull request by tacri**

### CODE REVIEW

File: server/dialogs/auth-dialog.js [hunk 1]
```diff
@@ -125,7 +129,10 @@ class AuthDialog extends builder.IntentDialog {
       let event = messageAsAny.originalInvoke;
       if (event.name === "signin/verifyState") {
         await this.handleLoginCallback(session);
-      } else {
+      } else if (event.name === "signin/tokenExchange") {
+        await this.handleTokenCallback(session);
+      }
+      else {
```
Comment (ydogandjiev) [inline server/dialogs/auth-dialog.js:135]: NIT: Put the `} else {` all on a single line

File: server/dialogs/auth-dialog.js [hunk 2]
```diff
@@ -166,6 +173,42 @@ class AuthDialog extends builder.IntentDialog {
...
+    const requestId = messageAsAny.originalInvoke.value.id;
+    const state = JSON.parse(utils.getOAuthState(session));
+
+    // check if there is a pending oauth request
+    if (state.securityToken == requestId) {
+      const pendingAction = state.pendingAction;
+
+      state.securityToken = state.pendingAction = undefined;
+      utils.setOAuthState(session, JSON.stringify(state));
+      let token = {
```
Comment (ydogandjiev) [inline server/dialogs/auth-dialog.js:188]: NIT: `let` --> `const`

File: manifest/sso-int/manifest.json [hunk 3]
```diff
@@ -60,6 +60,6 @@
   ],
   "webApplicationInfo": {
     "id": "ab93102c-869b-4d34-a921-a31d3e7f76ef",
-    "resource": "api://taskmeow.ngrok.io/ab93102c-869b-4d34-a921-a31d3e7f76ef"
+    "resource": "api://taskmeow.ngrok.io/botid-ab93102c-869b-4d34-a921-a31d3e7f76ef"
```
Comment (ydogandjiev) [inline manifest/sso-int/manifest.json:63]: What is the purpose of the `botid-` prefix here?

Comment (tacri) [reply manifest/sso-int/manifest.json:63]: we need to update the resource URI in of the AAD application to include the botid. Teams client is going to check that the bot requesting the token is the same as the one in the resource.

**Pull request merged**